conformance_tests/sysman/test_sysman_device/src/test_sysman_device_helper.cpp

```diff
@@ -12,21 +12,64 @@
+  [](const uint8_t(&arr)[ZE_MAX_DEVICE_UUID_SIZE]) {                           \
+    std::array<uint8_t, ZE_MAX_DEVICE_UUID_SIZE> uuid;                         \
+    std::copy(std::begin(arr), std::end(arr), uuid.begin());                   \
+    return uuid;                                                               \
+  }(x)
+
+void get_sysman_device_uuid(zes_device_handle_t sysman_device,
+                            std::vector<UUID> &sysman_device_uuids) {
+  zes_device_properties_t properties = {ZES_STRUCTURE_TYPE_DEVICE_PROPERTIES};
+  zes_device_ext_properties_t ext_properties = {
+      ZES_STRUCTURE_TYPE_DEVICE_EXT_PROPERTIES};
+  properties.pNext = &ext_properties;
+  EXPECT_EQ(ZE_RESULT_SUCCESS,
+            zesDeviceGetProperties(sysman_device, &properties));
+  auto sysman_device_uuid = ext_properties.uuid;
+  sysman_device_uuids.push_back(TO_STD_ARRAY(sysman_device_uuid.id));
+}
+
 void get_sysman_devices_uuids(std::vector<zes_device_handle_t> sysman_devices,
-                              std::vector<zes_uuid_t> &sysman_device_uuids) {
+                              std::vector<UUID> &sysman_device_uuids) {
+  for (const auto &sysman_device : sysman_devices) {
+    get_sysman_device_uuid(sysman_device, sysman_device_uuids);
+  }
+}
+
+void get_sysman_sub_devices_uuids(
+    std::vector<zes_device_handle_t> sysman_devices,
+    std::vector<UUID> &sysman_sub_device_uuids) {
   for (const auto &sysman_device : sysman_devices) {
-    zes_device_properties_t properties = {ZES_STRUCTURE_TYPE_DEVICE_PROPERTIES};
-    zes_device_ext_properties_t ext_properties = {
-        ZES_STRUCTURE_TYPE_DEVICE_EXT_PROPERTIES};
-    properties.pNext = &ext_properties;
-    ze_result_t result = zesDeviceGetProperties(sysman_device, &properties);
-    auto sysman_device_uuid = ext_properties.uuid;
-    sysman_device_uuids.push_back(sysman_device_uuid);
+    auto device_properties = lzt::get_sysman_device_properties(sysman_device);
+    uint32_t sub_devices_count = device_properties.numSubdevices;
+
+    if (sub_devices_count > 0) {
+      uint32_t num_sub_devices = 0;
+      auto sub_device_properties =
+          lzt::get_sysman_subdevice_properties(sysman_device, num_sub_devices);
+      EXPECT_EQ(sub_devices_count, num_sub_devices);
+      for (uint32_t sub_device_index = 0; sub_device_index < num_sub_devices;
+           sub_device_index++) {
+        EXPECT_LT(sub_device_properties[sub_device_index].subdeviceId,
+                  num_sub_devices);
+        EXPECT_GT(sub_device_properties[sub_device_index].uuid.id[0], 0);
+        sysman_sub_device_uuids.push_back(
+            TO_STD_ARRAY(sub_device_properties[sub_device_index].uuid.id));
+      }
+    } else {
+      // if subdevice doesn't exist for a device, then root device UUID
+      // is retrieved to match with core UUID's retrieved from flat mode
+      get_sysman_device_uuid(sysman_device, sysman_sub_device_uuids);
+    }
```

Comment:
Can we give this option to caller ? Caller will use get_sysman_devices_uuids method to get root device uuid's ?

Reply:
asking to use get_sysman_devices_uuids with extra parameter to get subdevice/root uuid ?

Reply:
Updated